core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -154,6 +154,113 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
   import org.bitcoins.core.psbt.InputPSBTRecord._
   import org.bitcoins.core.psbt.PSBTInputKeyId._
 
+  def nextRole(txIn: TransactionInput): PSBTRole = {
+    if (isFinalized) {
+      PSBTRole.ExtractorPSBTRole
+    } else {
+      (nonWitnessOrUnknownUTXOOpt, witnessUTXOOpt) match {
+        case (None, None) =>
+          PSBTRole.UpdaterPSBTRole
+        case (Some(_), None) | (None, Some(_)) | (Some(_), Some(_)) =>
+          val finalizeT = finalize(txIn)
+          finalizeT match {
+            case Failure(_) =>
+              val dummySigner = Sign.dummySign(ECPublicKey.freshPublicKey)
```

Comment:
What does this case mean? comments would be helpful, it seems strange to me we are using a dummy sign here.